java/impl/aws/src/main/java/co/worklytics/psoxy/aws/AwsModule.java

```diff
@@ -89,60 +83,84 @@ static CognitoIdentityClient cognitoClient(AwsEnvironment awsEnvironment) {
             .map(AwsEnvironment.SecretStoreImplementations::valueOf)
             .orElse(AwsEnvironment.SecretStoreImplementations.AWS_SSM_PARAMETER_STORE);
 
-        ConfigService sharedConfigService;
-        ConfigService instanceConfigService;
+        SecretStore sharedConfigService;
+        SecretStore instanceConfigService;
         if (secretStoreImpl == AwsEnvironment.SecretStoreImplementations.AWS_SECRETS_MANAGER) {
-            instanceConfigService = CompositeConfigService.builder()
-                .preferred(instanceScopedSecretsManagerConfigService)
-                .fallback(instanceScopedParameterConfigService)
-                .build();
-            sharedConfigService = CompositeConfigService.builder()
-                .preferred(secretsManagerConfigServiceFactory.create(pathToSharedConfig))
-                .fallback(sharedParameterConfigService)
-                .build();
-        } else if (secretStoreImpl == AwsEnvironment.SecretStoreImplementations.AWS_SSM_PARAMETER_STORE
-                || secretStoreImpl == AwsEnvironment.SecretStoreImplementations.HASHICORP_VAULT) {
-            /**
-             * Vault is wrapped over the 'Native' secrets manager in FunctionRuntimeModule
-             *
-             * TODO: probably consolidate this all in v0.5, when we eliminate fallback configs
-             * (where values in env vars take precedence over those in remote config stores;
-             *  and remote config store may be composite of parameter store + secrets manager)
-             *
-             * @see co.worklytics.psoxy.FunctionRuntimeModule
-             */
+            instanceConfigService = instanceScopedSecretsManagerConfigService;
+            sharedConfigService = secretsManagerConfigServiceFactory.create(pathToSharedConfig);
+        } else if (secretStoreImpl == AwsEnvironment.SecretStoreImplementations.AWS_SSM_PARAMETER_STORE) {
             instanceConfigService = instanceScopedParameterConfigService;
-            sharedConfigService = sharedParameterConfigService;
+            sharedConfigService = parameterStoreConfigServiceFactory.create(pathToSharedConfig);
+        } else if (secretStoreImpl == AwsEnvironment.SecretStoreImplementations.HASHICORP_VAULT) {
+            if (!vaultConfigServiceFactory.isVaultConfigured(envVarsConfigService)) {
+                throw new IllegalStateException("Vault is not configured, but HASHICORP_VAULT secret store implementation was requested");
+            }
+            VaultConfigService sharedVault =
+                vaultConfigServiceFactory.create(VaultConfigServiceFactory.pathForSharedVault(hostEnvironment, envVarsConfigService));
+
+            VaultConfigService connectorVault =
+                vaultConfigServiceFactory.create(VaultConfigServiceFactory.pathForInstanceVault(hostEnvironment, envVarsConfigService));
+
+            //initialize vaults (q: right place for this??)
```

Comment:
I think the rest of the comment is clear; not ideal but the alternative is more complex and ATM we don't need it